syntax = "proto3";

package apollo.fuelation;

import "fueling/learning/autotuner/proto/dynamic_model_info.proto";
import "fueling/learning/autotuner/proto/git_info.proto";

service SimService {
  rpc Initialize (InitParam) returns (Status) {}
  rpc RunScenario (JobInfo) returns (Status) {}
  rpc TearDown (Token) returns (Status) {}
}

message InitParam {
  GitInfo git_info = 1;
  int32 num_workers = 2;
  string token = 3;
  DynamicModel dynamic_model = 4;
}

message JobInfo {
  string token = 1;
  int32 scenario = 2;  // number indicating which worldsim scenario should be run
  map<string, string> model_config = 3;  // map from local config file path to serialized config
  string record_output_path = 4;  // storage path for output record file
  string record_output_file = 5;  // storage file name for output record file
  string iteration_id = 6;  // used to identify which training iteration it belongs to
}

message Token {
  string token = 1;
}

message Status {
  int32 code = 1;
  string message = 2;
}
